File: src/main/scala/io/izzel/taboolib/util/item/inventory/stored/ActionKeyboard.java
package io.izzel.taboolib.util.item.inventory.stored;

import io.izzel.taboolib.util.item.inventory.ClickEvent;
import org.bukkit.inventory.ItemStack;

/**
 * @Author sky
 * @Since 2019-12-03 19:20
 */
public class ActionKeyboard extends Action {

    @Override
    public ItemStack getCurrent(ClickEvent e) {
        return e.getClicker().getInventory().getItem(e.castClick().getHotbarButton());
    }

    @Override
    public void setCurrent(ClickEvent e, ItemStack item) {
        e.getClicker().getInventory().setItem(e.castClick().getHotbarButton(), item);
    }

    @Override
    public int getCurrentSlot(ClickEvent e) {
        return e.getRawSlot();
    }
}

File: module/bukkit-ui/src/main/kotlin/taboolib/module/ui/type/ActionKeyboard.kt
package taboolib.module.ui.type

import org.bukkit.inventory.ItemStack
import taboolib.module.ui.ClickEvent

class ActionKeyboard : Action() {

    override fun getCursor(e: ClickEvent): ItemStack? {
        return e.clicker.inventory.getItem(e.clickEvent().hotbarButton)
    }

    override fun setCursor(e: ClickEvent, item: ItemStack?) {
        e.clicker.inventory.setItem(e.clickEvent().hotbarButton, item)
    }

    override fun getCurrentSlot(e: ClickEvent): Int {
        return e.rawSlot
    }
}